python/postprocessing/modules/jme/jetmetUncertainties.py

```diff
@@ -447,10 +515,14 @@ def resolution_matching(jet, genjet):
               # apply v2 recipe correction factor also to JER central value and variations
               met_px_jer += delta_x_rawJet - met_unclEE_x
               met_py_jer += delta_y_rawJet - met_unclEE_y
-              met_px_jerUp += delta_x_rawJet - met_unclEE_x
-              met_py_jerUp += delta_y_rawJet - met_unclEE_y
-              met_px_jerDown += delta_x_rawJet - met_unclEE_x
-              met_py_jerDown += delta_y_rawJet - met_unclEE_y
+              
+              # FIXME is this correct?
+              for jerID in self.splitJERIDs:
+                  met_px_jerUp[jerID] += delta_x_rawJet - met_unclEE_x
+                  met_py_jerUp[jerID] += delta_y_rawJet - met_unclEE_y
+                  met_px_jerDown[jerID] += delta_x_rawJet - met_unclEE_x
+                  met_py_jerDown[jerID] += delta_y_rawJet - met_unclEE_y
```

Comment:
This loop seems to be consistent with what the code did here before your changes, right? So the question is not so much whether your change is consistent, but whether the method was already consistent before?

Reply:
paging also @danbarto

Reply:
Right, I had misunderstood the recipe somehow, it should be fine as it is... (I'll remove the FIXME from the file)

Reply:
The v2 recipe part should in fact only run when you start from raw MET (so I guess what's now under redoJEC=True) - did you check that this is done correctly?

Reply:
Hm, looking over it again I think the `redoJEC` flag is pretty much useless here and could be removed entirely.

About the v2 recipe, I'm not 100% sure what's being done and whether it is correct, e.g. it depends on what's given as `globalTagProd` option, and that is never set when calling from `jetmetHelperRun2.py`...

But this PR should be consistent with what is currently there in any case.